app/commands/xp.js

```diff
@@ -0,0 +1,444 @@
+
+            //informing player current point balance
+            this.tellPlayerRaw(playerName, [
+                { text: `You have retrieved `, color: 'white' },
+                { text: `${pointsRetrievedAll}`, color: 'red' },
+                { text: ` experience points.\nYou have `, color: 'white' },
+                { text: `0`, color: 'red' },
+                { text: ` points remaining.`, color: 'white' },
+            ]);
+
+        } else {
+            //make args[0] an integer
+            let argsInt = parseInt(args[0], 10);
+            
+            if (argsInt >= 0) {
+                //need to handle when player inputs more xp than they have xp stored
+                if (argsInt <= currentBalance) {
+                    //input less than or equal the xp stored.
+
+                    //adding the xp equal to number input as argsInt
+                    let pointsRetrievedPartial = await this.addPlayerExperience(playerName, argsInt);
+
+                    //removing those points from their xp store
+                    await this.updatePlayerXpStore(playerName, currentBalance - pointsRetrievedPartial);
+
+                    //informing player current point balance
+                    this.tellPlayerRaw(playerName, [
+                        { text: `You have retrieved `, color: 'white' },
+                        { text: `${pointsRetrievedPartial}`, color: 'red' },
+                        { text: ` experience points.`, color: 'white' },
+                    ]);
+                    await this.simpleCheck(playerName);
+
+                    
+                } else {
+                    //input too much xp.
+
+                    //just retrieving all thier xp.
+                    let pointsRetrievedInsufficient = await this.addPlayerExperience(playerName, currentBalance);
+
+                    //removing those points from their xp store
+                    await this.updatePlayerXpStore(playerName, currentBalance - pointsRetrievedInsufficient);
+
+                    //informing player current point balance
+                    this.tellPlayerRaw(playerName, [
+                        { text: `Not enough stored experience points.\nRetrieving `, color: 'white' },
+                        { text: `${pointsRetrievedInsufficient}`, color: 'red' },
+                        { text: ` experience points instead.`, color: 'white' },
+                    ]);
+                    await this.simpleCheck(playerName);
+                }
+
+            } else {
+                //They got here because they messed up
+                this.tellPlayerRaw(playerName, [
+                    { text: `Must input a positive number.\n`, color: 'red' },
+                    { text: `!xp get 50`, color: 'green' },
+                    { text: ` would retrieve 50 experience points.`, color: 'red' },
+                ]);
+            }
+        }
+    }
+
+    async handleXpGive(playerName, args) {
```

Comment:
When you know what the args are supposed to be name them the only time we should be seeing args is if it ambigious try something like

`async handleXpGive(playerName, [giftRecipient, amount]) {`

lots of functions in this file is using args.. if you can name the variable do so

Reply:
Changing my code to reflect this, but you should know I pretty much copied the format you used in warp and location and whatnot and thats why i wrote it that way.